src/main/generic/network/NetworkAgent.js

```diff
@@ -199,7 +201,8 @@ class NetworkAgent extends Observable {
             this._channel,
             msg.version,
             msg.startHeight,
-            msg.totalWork
+            msg.totalWork,
+            currentTimestampUTC - peerAddress.timestamp
```

Comment:
It's not very obvious why to take the timestamp of the address, maybe there should be some small comment on that (explicitly stating that this peer address object was created by the peer while sending the version message and thus the timestamp represents their local timestamp)